### PR TITLE
rename type "task" to "to-do"

### PR DESCRIPTION
**Pull request by mikez**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,20 +32,20 @@ $ git clone https://github.com/thingsapi/things.py && cd things.py && make insta
 
 ```python
 >>> import things
->>> things.tasks()
+>>> things.todos()
 [{'uuid': '2Ukg8I2nLukhyEM7wYiBeb',
   'title': 'Make reservation for dinner',
-  'type': 'task',
+  'type': 'to-do',
```
Comment (AlexanderWillner) [inline README.md:38]: In line 35 the method is called `todos`, here the type is called `to-do` - what about using the term `todo` everywhere just to be consistent?

File: things/database.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ def __init__(self, filepath=None):
     def get_tasks(
         self,
         uuid=None,
-        type="task",
+        type=None,
```
Comment (AlexanderWillner) [inline things/database.py:113]: while we're at it: the linters are not happy with a variable called `type`...

File: README.md (hunk 1 above)
Comment (mikez) [reply README.md:38]: The terminology used in the app is "to-do" and "to-dos", so it seemed that was the consistent term to use. Python has limitations on what characters one can use, so I had to drop the minus for the function name.

It is unclear to me whether the limitations of Python should supersede the naming used in the app. Or... what do you think?

File: things/database.py (hunk 2 above)
Comment (mikez) [reply things/database.py:113]: In my opinion, that's an ok a usage. We are prioritizing convenience of API usage and consistency of naming with the Things app here. Yes, it's true, there is a built-in function called `type`, but in my opinion it's also fine to overwrite things in Python when the local usage is a more fitting one.

Examples from the standard library: [socket](https://github.com/python/cpython/blob/c4d45ee670c09d4f6da709df072ec80cb7dfad22/Lib/socket.py#L220), [asyncio](https://github.com/python/cpython/blob/e3ef4d7f653976ac0ccacc4e3fde06bf0e0f139b/Lib/asyncio/events.py#L295), [argparse](https://github.com/python/cpython/blob/e3ef4d7f653976ac0ccacc4e3fde06bf0e0f139b/Lib/argparse.py#L820), [tokenize](https://github.com/python/cpython/blob/e3ef4d7f653976ac0ccacc4e3fde06bf0e0f139b/Lib/tokenize.py#L45)

**Comment by mikez** [comment]
@AlexanderWillner Here are some work opportunities if you'd enjoy playing around with the API:
1. write tests.
2. interactively query your Things database in iPython or Jupyter to get a feel for it.
3. implement `things.trashed()` – will require tweaking `Database.get_tasks`.
4. implement or improve any other utility function you think might be useful.

**Comment by AlexanderWillner** [comment]
Was thinking about porting the CLI to use this library.

**Comment by mikez** [comment]
👌

**Comment by AlexanderWillner** [comment]
Actually, you can find an early starting point at https://github.com/thingsapi/things-cli. I think I found the first issue:

```shell
% ./bin/things-cli today
 -  To-Do in Today  ( Anytime )
 -  To-Do in Upcoming  ( Someday )
 -  Repeating To-Do  ( Someday )
```

Probably this should only return a single entry.

**Comment by AlexanderWillner** [comment]
Never mind, this was already fixed in this merge request ;)

**Pull request merged**